metricbeat/module/elasticsearch/cluster_stats/data_xpack.go

```diff
@@ -0,0 +1,238 @@
+				"terms_memory_in_bytes":         c.Int("terms_memory_in_bytes"),
+				"stored_fields_memory_in_bytes": c.Int("stored_fields_memory_in_bytes"),
+				"term_vectors_memory_in_bytes":  c.Int("term_vectors_memory_in_bytes"),
+				"norms_memory_in_bytes":         c.Int("norms_memory_in_bytes"),
+				"points_memory_in_bytes":        c.Int("points_memory_in_bytes"),
+				"doc_values_memory_in_bytes":    c.Int("doc_values_memory_in_bytes"),
+
+				"index_writer_memory_in_bytes":  c.Int("index_writer_memory_in_bytes"),
+				"version_map_memory_in_bytes":   c.Int("version_map_memory_in_bytes"),
+				"fixed_bit_set_memory_in_bytes": c.Int("fixed_bit_set_memory_in_bytes"),
+				"max_unsafe_auto_id_timestamp":  c.Int("max_unsafe_auto_id_timestamp"),
+			}),
+		}),
+		"nodes": c.Dict("nodes", s.Schema{
+			"count": c.Dict("count", s.Schema{
+				"total":             c.Int("total"),
+				"data":              c.Int("data"),
+				"coordinating_only": c.Int("coordinating_only"),
+				"master":            c.Int("master"),
+				"ingest":            c.Int("ingest"),
+			}),
+			"os": c.Dict("os", s.Schema{
+				"available_processors": c.Int("available_processors"),
+				"allocated_processors": c.Int("allocated_processors"),
+				"mem": c.Dict("mem", s.Schema{
+					"total_in_bytes": c.Int("total_in_bytes"),
+					"free_in_bytes":  c.Int("free_in_bytes"),
+					"used_in_bytes":  c.Int("used_in_bytes"),
+					"free_percent":   c.Int("free_percent"),
+					"used_percent":   c.Int("used_percent"),
+				}),
+			}),
+			"process": c.Dict("process", s.Schema{
+				"cpu": c.Dict("cpu", s.Schema{
+					"percent": c.Int("percent"),
+				}),
+				"open_file_descriptors": c.Dict("open_file_descriptors", s.Schema{
+					"min": c.Int("min"),
+					"max": c.Int("max"),
+					"avg": c.Int("avg"),
+				}),
+			}),
+			"jvm": c.Dict("jvm", s.Schema{
+				"max_uptime_in_millis": c.Int("max_uptime_in_millis"),
+				"mem": c.Dict("mem", s.Schema{
+					"heap_used_in_bytes": c.Int("heap_used_in_bytes"),
+					"heap_max_in_bytes":  c.Int("heap_max_in_bytes"),
+				}),
+				"threads": c.Int("threads"),
+			}),
+			"fs": c.Dict("fs", s.Schema{
+				"total_in_bytes":     c.Int("total_in_bytes"),
+				"free_in_bytes":      c.Int("free_in_bytes"),
+				"available_in_bytes": c.Int("available_in_bytes"),
+			}),
+		}),
+	}
+)
+
+func passthruField(fieldPath string, sourceData, targetData common.MapStr) error {
+	fieldValue, err := sourceData.GetValue(fieldPath)
+	if err != nil {
+		return elastic.MakeErrorForMissingField(fieldPath, elastic.Elasticsearch)
+	}
+
+	targetData.Put(fieldPath, fieldValue)
+	return nil
+}
+
+func eventMappingXPack(r mb.ReporterV2, m *MetricSet, content []byte) error {
+	var data map[string]interface{}
+	err := json.Unmarshal(content, &data)
+	if err != nil {
+		return err
+	}
+
+	clusterStats, err := clusterStatsSchema.Apply(data)
+	if err != nil {
+		return err
+	}
+
+	dataMS := common.MapStr(data)
+
+	passthruFields := []string{
```

Comment:
In general it seems these are values which outside x-pack reporting should be reported by the specific metricsets and to get all names for example it would be an aggregation based on the cluster id?